2015-04-10-deploying-simple-html-projects-to-ghpages-using-gulp.markdown

```diff
@@ -0,0 +1,178 @@
+          .pipe(gulp.dest('./build/js'));
+    });
+
+> Here we added the pipe **uglify** to compress our JS code
+
+So now we know how to compile our **SASS** and **CoffeeScript** files. Everything is fine and dandy. What next?  
+How about serving these files? Sure.
+
+Gulp can use a bunch of packages to serve files. So far my favorite is **[BrowserSync](http://www.browsersync.io/)**
+
+    var gulp        = require('gulp'),
+        browserSync = require('browser-sync');
+
+    gulp.task('server', function () {
+      browserSync({
+        files: ['./build/js/*.js', './build/css/*.css', './build/*.html'],
+        server: {
+          baseDir: './build'
+        }
+      });
+    });
+
+Then in your terminal
+
+    $ gulp server
+
+This will start the server on port 3000.  
+Observe that in the line `files: ['./build/js/*.js', './build/css/*.css', './build/*.html']` **BrowserSync** will keep an eye on these files so in case they change the server will automatically reload them.
+
+But this only starts the server, you still need to compile and copy all files to the `./build` folder.
+
+> "So I'll have to run each tasks every time I want to start the server?"
+
+![what-if-i-told-you](http://i.imgur.com/XOCkn7k.jpg)
+
+We can create tasks that can execute others tasks in a row. So let's take the tasks we created before and put them together under another task
+
+    gulp.task('default', ['sass', 'scripts', 'server']);
+
+To run a `default` task you just need to ender `gulp` in the terminal. This will run sequentially the `sass`, `scripts` and `server` tasks. So now you have your server running with all files properly compiled.
+
+**One more thing.**
+Remember when we create the `server` task and we made the server to watch some files so when they change they would automatically be reloaded? Well this won't work the way you imagine. You see the path of those files are `./build/` while you will be working on the folder `./src/`. So when you update your `./src/css/main.css.scss` this will mean nothing to the server. You would have to restart the server to recompile these assets. Which means that line is complete useless, right? No. Not actually.
+
+We just need to create another task that will watch our files on `./src/`. And it goes like this:
+
+    gulp.task('watch', function() {
+      gulp.watch('./src/css/**/*.scss', ['sass']);
+      gulp.watch('./src/js/**/*.coffee', ['scripts']);
+    });
+
+Then we update our `default` task to:
+
+    gulp.task('default', ['sass', 'scripts', 'server', 'watch']);
+
+Now right after the server starts Gulp will watch the files we told him. So for example, when we change our `./src/css/main.css.scss`, Gulp will execute the `sass` task and recompile the file `./build/css/main.css` that our server is watching and will reload it (Actually BrowserSync will inject the changes into the main.css that are already loaded on the browser without even reloading the browser. Told you guys BrowserSync is awesome).

```

Comment:
I'd add a HUGE MAGIC GIF HERE... this is really the best thing ever in frontend development's world.